<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>MOFA Integration</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            line-height: 1.6;
            margin: 20px;
        }

        h2, h3, h4 {
            color: #3498db;
        }

        .center {
            text-align: center;
            margin-top: 20px;
            margin-bottom: 20px;
        }

        .center img {
            display: block;
            margin: 0 auto;
        }

        ol {
            list-style-type: decimal;
            padding-left: 20px;
        }

        .banner {
            background-color: #3498db; /* Blue background color */
            padding: 10px;
            text-align: left;
            color: white;
        }

        .banner a {
            color: white;
            margin-right: 15px;
            text-decoration: none;
        }
    </style>
</head>

<body>

    <!-- Banner with Navigation Links -->
    <div class="banner">
       <a href="index.html">BiomiX</a>
       <a href="Installation.html">Installation</a>
       <a href="Tutorial.html">Tutorial</a>
       <a href="Single-omics_analysis.html">Single-omics analysis</a>
       <a href="Integration_analysis.html">Integration</a>
       <a href="Parameters_and_interface.html">Parameters and interface</a>
       <a href="about_us.html">About us</a>
    </div>

    <h2 id="Omics and integration principles">Omics-and-integration-principles</h2>

    <p>To provide the users with all the tools to understand each omics data and how to analyse them we wrote this detailed guide to address the users to the sources where understand the principles of each omics.</p>

    <h2 id="Transcriptomics">Transcriptomics:</h2>

    

    Bulk RNA sequencing (RNA-seq) is a widely used technique to analyze the transcriptome, which represents the complete set of RNA molecules in a population of cells. It provides insights into gene expression, splicing events, and transcript abundance across different conditions or treatments. Common steps include:

  <ol>
    <li><strong>RNA Isolation:</strong> RNA is extracted from the sample, typically using methods that preserve the integrity of the RNA. The RNA Integrity Number (RIN) is the value to estimate the RNA integrity. A score above 6-7, on a RIN scale from 0 to 10 is required to asses a good RNA quality.</li>
    <li><strong>cDNA Library Preparation and sequencing:</strong> The RNA is converted into complementary DNA (cDNA) using reverse transcription. The cDNA is then fragmented, and adapters are added to facilitate sequencing. the library is sequenced on platforms such as Illumina, generating millions of short nucleotide sequences called reads. These reads represent fragments of the original RNA molecules. This video illustrates the method: <a href="https://www.youtube.com/watch?v=fCd6B5HRaZ8&ab_channel=Illumina">Link</a></li>
    <li><strong>Mapping and Matrix Data Generation:</strong> The sequencing reads are mapped to a reference genome or transcriptome using specialized software like STAR, HISAT2, or Bowtie2. Mapping aligns the reads to specific regions, helping identify which genes or transcripts are expressed. Once mapped, the number of reads aligning to each gene is counted, generating a count matrix. Each row represents a gene, and each column corresponds to a sample. The count matrix is the primary data format used for downstream analysis.</li>
    
    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Detailed_1.png?raw=true" width="900" height="600">
    </div>
    
    <li><strong>Statistical Properties of RNA-seq Data:</strong> RNA-seq count data is not normally distributed but follows discrete distributions, often resembling a negative binomial distribution due to overdispersion (variability greater than expected under a Poisson model). The data may have large dynamic ranges (from very low to very high expression, defined as <strong> “homoscedasticity” </strong> ), and sequencing depth variability ( <strong> “Size factor” </strong> ) that must be accounted for during analysis. </li>   

    -DESeq2: This R packages model RNA-seq count data using a negative binomial distribution. It performs Size factor normalization and statistical testing to account for variability in sequencing depth and dispersion. By default, in BiomiX the counts are normalized by variance stabilizing transformation (vst) to correct the homoscedasticity.
    -Limma: This package is suitable for normalized data that follow an approximately normal (Gaussian) distribution. Examples (RPKM, FPKM and TPM, vst normalized).

    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Detailed_2.png?raw=true" width="900" height="600">
    </div>
  </ol>


    
    <h2 id="Metabolomics">Metabolomics:</h2>

    <p>Metabolomics is the comprehensive study of metabolites—small molecules involved in metabolism—within a biological system. These metabolites reflect the biochemical activities of cells, tissues, or organisms and provide insights into their physiological and pathological states. Here is an explanation of the principles: <a href="https://www.youtube.com/watch?v=D77pACc0uF8&ab_channel=Chemistorian">Link</a></p>
    <p>Metabolomics typically involves the extraction of metabolites from biological samples (e.g., tissues, blood, or urine), followed by analysis using high-throughput techniques such as mass spectrometry (MS) or nuclear magnetic resonance (NMR) spectroscopy. These methods measure the abundance of metabolites, enabling their identification and quantification. Common steps in this analysis include:</p
                                                                                                                                                                                                                                                                                                                                                                                                                    
    <li><strong>Data generation and Peak Detection::</strong> Identifying metabolite signals in raw data.</li>
    <li><strong>Feature Alignment:</strong> Aligning signals across samples to ensure consistency.</li>
    <li><strong>Normalization:</strong> Correcting for variations in sample processing or instrument performance. </li>

     These steps are not included in BiomiX, but a guideline of how perform these steps is reported here: 
     doi: 10.3390/metabo10010028
     <a href="https://doi.org/10.3390/metabo10010028">Literature reference</a>
     <a href="https://www.metaboanalyst.ca/docs/RTutorial.xhtml#2.1%20LC-MS/MS%20Raw%20Spectra%20Processing">Metaboanalyst Raw Spectra Processing</a>
     <a href="https://api2.xialab.ca/api/download/metaboanalyst/1_LC_MS_Spectra_Processing.pdf">Metaboanalyst Raw Spectra Processing pdf</a>

    <li><strong>Annotation and Identification: </strong> Assigning chemical identities to detected features using databases like HMDB (Human Metabolome Database) or KEGG. </li>
    <li><strong>Data Distribution and Statistical Analysis: </strong> Metabolomic data often exhibit non-normal distributions and require appropriate statistical tools </li>
    <li><strong>Pathway Analysis: </strong> Tools like MetaboAnalyst link metabolites to biological pathways for functional insights </li>

    

    <h3>Targeted vs. Untargeted Metabolomics</h3>

    <h4>Targeted Metabolomics</h4>
    <p>In targeted metabolomics, researchers focus on a predefined set of metabolites, often selected based on prior knowledge or specific hypotheses.</p>
    <ul>
        <li><strong>Advantages:</strong> High sensitivity, precision, and quantification accuracy.</li>
        <li><strong>Applications:</strong> Quantitative analysis of specific pathways or biomarkers, such as glucose levels in diabetes research.</li>
        <li><strong>Limitations:</strong> Restricted to a limited number of metabolites, potentially missing novel or unexpected findings.</li>
    </ul>

    <h4>Untargeted Metabolomics</h4>
    <p>Untargeted metabolomics aims to detect as many metabolites as possible in a sample, providing an unbiased snapshot of the metabolic state.</p>
    <ul>
        <li><strong>Advantages:</strong> Comprehensive analysis, enabling the discovery of unknown metabolites and broad profiling of pathways.</li>
        <li><strong>Applications:</strong> Biomarker discovery, hypothesis generation, and exploration of novel biochemical processes.</li>
        <li><strong>Limitations:</strong> Lower precision for quantification, challenges in identifying unknown metabolites, and greater complexity in data analysis.</li>
    </ul>
    <p><strong>Source image:</strong> <a href="https://www.drugtargetreview.com/article/33033/the-next-revolution-in-stratified-medicine-molecular-phenotyping-with-metabolomics/" target="_blank">https://www.drugtargetreview.com/article/33033/the-next-revolution-in-stratified-medicine-molecular-phenotyping-with-metabolomics/</a></p>

    <h2>Methylomics</h2>
    <p>Methylomics involves studying DNA methylation, an essential epigenetic modification where methyl groups are added to cytosines, typically in CpG dinucleotides. This modification regulates gene expression and plays a crucial role in development, cellular differentiation, and disease processes such as cancer. The Illumina Infinium HumanMethylation450 BeadChip (450K) and MethylationEPIC BeadChip (EPIC) arrays are widely used platforms for profiling genome-wide DNA methylation.</p>
    <p><strong>Principles of the techniques:</strong> <a href="https://www.youtube.com/watch?v=lVG04dAAyvY&ab_channel=Illumina" target="_blank">Link</a></p>
    <p>Common Steps:</p>
    <ul>
        <li><strong>Data generation:</strong> Both arrays require bisulfite-converted DNA (~500 ng per sample), where unmethylated cytosines are converted to uracil while methylated cytosines remain unchanged.</li>
    </ul>

    <h3>450K Array</h3>
    <ul>
        <li><strong>Coverage:</strong> The 450K array targets ~485,000 CpG sites, including promoters, gene bodies, and intergenic regions.</li>
        <li><strong>Target Regions:</strong> Cover CpG islands, shores, shelves, and open seas.</li>
    </ul>

    <h3>EPIC Array</h3>
    <ul>
        <li><strong>Coverage:</strong> The EPIC array extends this coverage to >850,000 CpG sites, adding enhancers and regulatory elements identified by projects like ENCODE and FANTOM.</li>
        <li><strong>Target Regions:</strong> Both arrays cover CpG islands, shores, shelves, and open seas and include additional enhancer regions and non-CpG methylation sites, improving its utility for diverse studies.</li>
    </ul>

    <h2>Output Data</h2>
    <p>The arrays generate beta values (ranging from 0 to 1), representing the fraction of methylation at each CpG site. M-values can also be provided. Before using them in BiomiX, we suggest converting them in R:</p>
    <pre>
        <code>
        → Open R studio
        if (!require("BiocManager", quietly = TRUE))
            install.packages("BiocManager")
        BiocManager::install("lumi")

        library(lumi)
        m <- read.table("path/to/the/matrix")
        m <- m2beta(m)
        write.table(m, file="Matrix_converted_to_beta_value", sep = "\t", quote = FALSE, row.names = FALSE)
        </code>
    </pre>
    <p><strong>Detection p-values:</strong> These ensure the reliability of each site, and poor-quality data can be filtered out.</p>



    <h2 id="Main interface guide">Main interface guide</h2>

    <p>The main interface structure:</p>
    <li><strong>Condition.</strong> Group to analyse.</li>
    <li><strong>Control.</strong> Reference group, compared with the condition one.</li>
    <li><strong>Output.</strong> Directory where save the output results of the analysis.</li>
    <p>Omics input grid:</p>
    <li><strong>Input.</strong> Input slot number.</li>
    <li><strong>Preview-QC.</strong> Opens a shiny app that visualizes, transforms the data and removes variables or sample outliers.</li>
    <li><strong>Single omics Analysis.</strong> Check the box to analyse that omic by our single omics pipelines.</li>
    <li><strong>Data type.</strong> Select the type of omics uploaded.</li>
    <li><strong>Integration.</strong> Check the box to include those omics in the MOFA integration.</li>
    <li><strong>Label.</strong> Omics name, used to name the output folder and filter the sample based on the sample names.</li>
    <li><strong>Selection.</strong> If selected filters the samples based on the sample name using a regex derived by the label name.</li>
    <li><strong>Data upload.</strong> Click on the button to upload the matrix. Here, it is possible to modify the BiomiX format (.tsv,xls,xlsx) matrix. </li>
    <p>Integration grid:</p>
    <li><strong>Integration.</strong> Do you want to perform the integration with the uploaded omics data?.</li>
    <li><strong>Method.</strong> Which integration method? (Only MOFA is available for now).</li>
    <li><strong>N° Factors.</strong> How many MOFA factors do you want to calculate for the integration? (select 0 for an automatic selection).</li>
    <li><strong>Factor to explore.</strong> Which factor do you want to explore graphically? (Contributors, heatmap clustering etc...).</li>
    <li><strong>Omics overlay.</strong> What is the minimum number of omics that a sample should have to be included in the integration?</li>
    <li><strong>Open advance option.</strong> Button to open the advance option interface</li>
    <li><strong>Open BiomiX chatbot.</strong> Button to open the BiomiX chatbot</li>
    <li><strong>Start Analysis.</strong> Button to start the analysis.</li>
    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/MAIN_INTERFACE_1.svg?raw=true">
    </div>
    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/MAIN_INTERFACE_2.svg?raw=true">
    </div>

    <h2 id="Guide to the advanced option interface">Guide to the advanced option interface</h2>

    <h3 id="General">Advanced options interface (General section):</h3>
    <li><strong>Log2FC threshold.</strong> Log2FC threshold value for significant results.</li>
    <li><strong>P.adj threshold.</strong> Adjusted p.value threshold value for significant results.</li>
    <li><strong>Gene Panel.</strong> Button to upload the gene panel for the subgrouping analysis.</li>
    <li><strong>Array type.</strong> The type of chip from which the data comes (450K/EPIC).</li>
    <li><strong>N° of genes within the panel with score > one or two.</strong> Parameters to define the positive/negative subgrouping by the gene panel. The user can modify the number of genes that must have a Z-score > one or two of the control standard deviation</li>
    <li><strong>Removal of positive control samples.</strong> If checked, the controls positive for the gene panel can be excluded from the downstream analysis.</li>
    <li><strong>N° top DE genes in heatmap</strong> Number of top genes(or metabolites) to visualize in the heatmaps.</li>
    <li><strong>Clustering distance.</strong> Clustering distance used in the heatmaps, both in subgrouping and single omics analysis.</li>
    <li><strong>Clustering methods.</strong> Clustering method used in the heatmaps, both in subgrouping and single omics analysis.</li>
    <li><strong>CPU threads.</strong> Number of CPU threads used in your PC in parallel.</li>
    <li><strong>N° MOFA input features.</strong> Number of top features used in the MOFA analysis. It is suggested to use a similar number of features in each omics</li>
    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/General.png?raw=true" width="900" height="600">
    </div>

    <h3 id="Metabolomics">Advanced options interface (Metabolomics section):</h3>
    <li><strong>Metabolite annotation.</strong> Which annotation (HMDB, KEGG, compound name) are used in the uploaded annotated metabolomics matrices.</li>
    <li><strong>Ion mode.</strong> Type of ionization used in the metabolomics data acquisition.</li>
    <li><strong>M/Z Tolerance ppm.</strong> Ppm tolerance used in the MS1 annotation.</li>
    <li><strong>Adduct positive mode.</strong> Type of positive adduct generated during the metabolomics data acquisition.</li>
    <li><strong>Adduct negative mode.</strong> Type of negative adduct generated during the metabolomics data acquisition.</li>
    <li><strong>Adduct neutral mode.</strong> If checked will consider these data as neutral.</li>
    <li><strong>MS1 files upload.</strong> Button were upload the MS1 annotation (required also for the MS2) with the corresponding input number.</li>
    <li><strong>Databases MS1.</strong> Databases consulted by the MS1 annotation (By Ceu Mass Mediator).</li>
    <li><strong>mz match MS1/2.</strong> Mass charge ratio tolerance for the match between MS1 annotation and the fragmentation spectra MS2 provided within the .mzml or .mgf files.</li>
    <li><strong>RT match MS1/2.</strong> Retention time tolerance for the match between MS1 annotation and the fragmentation spectra MS2 provided within the .mzml or .mgf files.</li>
    <li><strong>Column.</strong> Type of column used in the liquid chromatography.</li>
    <li><strong>Databases MS2.</strong>  Databases consulted by the MS2 annotation (By Tidymass).</li>
    <li><strong>MS2 directory</strong> Directory containing the fragmentation spectra (.mzml or .mgf)</li>
    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Advanced_1.svg?raw=true" width="900" height="600">
    </div>
    <h3 id="Metadata">Advanced options interface (Metadata section):</h3>
    <li><strong>Column name</strong> Column used for the sample filtering</li>
    <li><strong>Column name</strong> Type of data in the defined column</li>
    <li><strong>Threshold/Factor selected</strong> Threshold to filter the samples (format: ">= 90" / "== 0.56") or specific condition (format: "== male" / "== treated") </li>
    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Metadata.png?raw=true" width="900" height="600">
    </div>

    <h3 id="MOFA">Advanced options interface (MOFA section):</h3>
    <li><strong>Max iteration.</strong> Maximum number of MOFA iterations in the model training.</li>
    <li><strong>Convergence mode.</strong> Speed to train the MOFA model. The longer the more accurate.</li>
    <li><strong>Threshold contribution weight.</strong> Threshold to isolate the top MOFA factor contributors. Among the 5% top percent contributors (prefiltered by BiomiX) the user can define a contribution threshold set to 0.5 by default. </li>
    <li><strong>Type of research (Bibliography).</strong> The type of document where the contributors will be researched in Pubmed. Abstract/Title or Text word.</li>
    <li><strong>N° articles (Bibliography).</strong> Number of articles consulted by BiomiX to select articles related to the significant factors. Attention, Pubmed server has a max number of request and the research of too many contributors in too many articles can drive the request interruption with the server. In this case reduce the number of articles consulted or number of contributors researched.</li>
    <li><strong>N° top contributors (Bibliography).</strong> Top contributors of each omics researched in the PubMed abstracts.</li>
    <li><strong>N° keywords extracted (Bibliography).</strong> Number of keywords extracted in each article related to significant MOFA factors. These are the keywords actively extracted selecting the most cited words of the abstract by natural language processing.</li>
    <li><strong>P.adj threshold (Pathway mining).</strong> Adjusted p.value threshold to consider significant a biological pathway.</li>
    <li><strong>Pathways shown (Pathway mining).</strong> Number of biological pathways visualized as output in the pdf reports.</li>
    <li><strong>Numerical (Clinical).</strong> Check the box to correlate the clinical data (Numeric) with the significant MOFA factor.</li>
    <li><strong>Binary (Clinical).</strong> Check the box to correlate the clinical data (Binary) with the significant MOFA factor.</li>
    <li><strong>Save advanced options (Save parameters).</strong> Click on the blue button to save the advanced parameters.</li>
    <br>
    <br>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/MOFA.png?raw=true" width="900" height="600">
    </div>

</body>

</html>
