<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Single-omics Analysis</title>
    <style>
        body {
            font-family: 'Lato', sans-serif;
            line-height: 1.6;
            margin: 20px;
        }

        .banner {
            background-color: #3498db;
            padding: 10px;
            text-align: left;
            color: white;
        }

        .banner a {
            color: white;
            margin-right: 15px;
            text-decoration: none;
        }

        h2, h3, h4 {
            color: #3498db;
        }

        img {
            max-width: 100%;
            height: auto;
            margin: 20px 0;
        }

        .additional-features {
            margin-left: 20px;
        }

        .copy-button {
            background-color: #3498db;
            color: white;
            border: none;
            padding: 5px 10px;
            text-align: center;
            text-decoration: none;
            display: inline-block;
            font-size: 14px;
            margin-bottom: 10px;
            cursor: pointer;
            border-radius: 3px;
        }
    </style>
</head>

<body>
    
    <!-- Banner with Navigation Links -->
    <div class="banner">
       <a href="index.html">BiomiX</a>
       <a href="Installation.html">Installation</a>
       <a href="Tutorial.html">Tutorial</a> 
       <a href="Single-omics_analysis.html">Single-omics analysis</a>
       <a href="Integration_analysis.html">Integration</a>
       <a href="Biological_interpretation.html">Biological Interpretation</a>
       <a href="about_us.html">About us</a>
    </div>

    <h3 id="General-guidelines">General guidelines</h3>
    <p>BiomiX was developed to compare two groups using single and multiple omics. To work properly, the input file must follow a specific format and criteria.</p>
    <p><strong>Metadata: </strong></p>
    <p>BiomiX requires uploading a metadata file in the BiomiX launcher. The first column must contain the sample ID, while the others contain other sample variables.</p>
    <li><strong>"ID" column (mandatory):</strong> This is the first column and contains the sample ID.</li>
    <li><strong>"CONDITION" column (mandatory):</strong> This column must contain the sample condition. These conditions will be available for selection in the main interface.</li>
    <li><strong>"SAMPLE_TYPE" column (optional):</strong> This column is designed for specific conditions. Suppose the input matrix contains samples from different tissues or cells (convoluted matrix), and this source is also defined in the sample ID (e.g., ID = 25658_BLymphocyte, 25658_GlialCells). In that case, this column allows analysis to be performed only on data from the selected tissues or cells.</li>
    <li><strong>"GENDER" column (optional): </strong> This column can be used to adjust the LIMMA or DESeq2 model for transcriptomics data.</li>
    <li><strong>"AGE" column (optional):</strong> This column can also be used to adjust the LIMMA or DESeq2 model for transcriptomics data.</li>
 
<p>Any other metadata column can be used to filter the data via the Metadata section in the advanced options or for the PCA and UMAP preview-QC visualization.</p>
    
    <h3 id="Matrix"><p><strong>Matrix: </strong></p></h3>

    <p>BiomiX requires an expression matrix where the columns represent the samples and the rows contain the variables.</p>
    <p><strong>Omics differences: </strong></p>
      
    <li><strong>In transcriptomics,</strong> the variables (genes) should be named using GENE SYMBOL or ENSEMBL ID.</li>
    <li><strong>In metabolomics,</strong> variables (metabolites) should be named using KEGG, HMDB, or official compound names. If the peaks are not annotated, the format should be "peak_[number]," especially for MS1 and MS2 annotation.</li>
    <li><strong>In methylomics, </strong> variables must use the EPIC or 450K array code. </li>
    <li><strong>In Undefined omics, </strong> any variable name is accepted.</li>

    
    <h3 id="transcriptomics">Transcriptomics</h3>

    <h4 id="Single-Omics-Analysis">Single Omics Analysis</h4>
    <p>The transcriptomics single omics analysis identifies differentially expressed genes (DEGs) along with their statistics (Log2FC and adjusted p-values), and provides visualizations such as volcano plots and heatmaps. If <strong>SEX</strong> and <strong>GENDER</strong> columns are included in the metadata, they will be used to adjust the DESeq2 or Limma model. The default thresholds for DEGs are set to Log2FC > 0.5 and adjusted p-value < 0.05.</p>

    <p>Biological process enrichment is explored using the R version of EnrichR, with options to export data for external tools such as <strong>GSEA</strong> <a href="https://www.gsea-msigdb.org/gsea/downloads.jsp"> Link </a> and <strong>EnrichR</strong> <a href="https://maayanlab.cloud/Enrichr/"> Link </a>. Within the Preview-QC visualization, users can choose to transform the data: if transformed, DGE is performed with Limma; if not, and the expression matrix contains raw counts, DESeq2 is used instead. The transformed data are then used as input for MOFA integration. To optimize analysis speed, only the most variable genes are selected (5000 by default, customizable in the MOFA section of the advanced options).</p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Transcriptomics.svg?raw=true">
    </div>

    <div class="additional-features">
        <h4>Additional Features</h4>

        <ul>
            <li>
                <strong>Gene Panel subgrouping:</strong> It is possible to provide a panel of genes within <strong>GENE_PANEL</strong> file, in the <strong>General section of the advanced options</strong> The genes must be in Gene symbol format and tab-delimited (by enter button).
                <pre><code>GENES_FOR_SUBPOPULATION
SIGLEC1
IFIT3
IFI6
LY6E
MX1
USP18
OAS3
IFI44L
</code></pre>
               These genes will generate a heatmap displaying the expression standard deviation of each specific gene across samples. This enables classification of condition/disease samples into positive and negative groups, while BiomiX performs DGE analysis to compare the chosen controls with these two subgroups
            </li>
            <li>
            <strong>Clinical or Biological Marker Validation:</strong> Adding a column named <strong>MARKER</strong> in the metadata is possible to classify samples based on clinical or biological marker measurements. This marker classification can then be compared to gene panel subgrouping to observe similarities or differences.
            </li>
        </ul>

        <div align="center">
            <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Heatmap_BLymphocytes_CTRL_vs_SLE2.svg?raw=true">
        </div>

        <div align="center">
            <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/enrichR.svg?raw=true">
        </div>

        <div align="center">
            <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/GSEA2.svg?raw=true">
        </div>
    </div>

    <h3 id="metabolomics">Metabolomics</h3>

    <p>The matrix's peak or metabolite signals are first visualized by preview-QC to decide if and how to transform the data. The transformed signals are then compared among the chosen groups by calculating Log2FC as the log2 ratio of their median peak signals. P-values are calculated using the non-parametric Mann–Whitney test and corrected using the FDR method. The Volcano plot and heatmap display significant metabolites or peaks that are increased or decreased. <strong>Warning:</strong> Please ensure the data matrix is curated to remove variables without biological relevance (e.g., contaminants, poorly reproducible signals) and that instrumental normalization is performed before uploading to BiomiX. All transformed peak or metabolite signals are included for MOFA integration.</p>

    <p>To reveal changes in biological pathways, BiomiX uses methPath v1.0.5 from tidymass v1.0.8 to detect enrichment in metabolic processes among significantly changed metabolites. For deeper analysis, it also prepares ready-to-copy files for input into MetaboAnalyst, one of the most advanced online tools for metabolomics analysis, supporting enrichment and metabolite set enrichment analysis. If transcriptomic results are available in BiomiX, the platform will automatically integrate metabolomics and transcriptomics data to create files ready for MetaboAnalyst's joint pathway and network analysis.</p>

    <p>If you have only raw .mzML files, we strongly recommend using the R pipeline by Á. Fernández-Ochoa (https://doi.org/10.3390/metabo10010028) or MetaboAnalyst to generate the peak signals matrix.</p>


    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Metabolomics.svg?raw=true">
    </div>

    <div class="additional-features">
        <h4>Additional Features</h4>

        <p>BiomiX is flexible, supporting the analysis of both targeted (annotated) and untargeted metabolomics data, with the option to analyze untargeted data with or without annotation. The untargeted annotation is based on MS1 annotation (mass/charge ratio “m/z” and retention time values) or MS2 data (MS1 annotation plus raw MS2 fragmentation files in mzML or .mgf format).</p>

        <ul>
            <li>
                <strong>Peaks automatic annotation:</strong>
                <p><i>-Targeted metabolomics:</i> This option supports metabolomics data obtained from any analytical platform such as LC-MS, GC-MS, CE-MS or NMR, provided the metabolites' biological identities are available. Despite the annotation here is not performed the choice of the proper metabolite identifier (HMDB or KEGG) is required for biological pathway analysis. Also the Iupac compound name option is available but will not perform any biological pathway analysis.</p>
                <p><i>-MS1 annotation (Untargeted HRMS: High-resolution mass spectrometry):</i> Users can upload MS1 files, containing mass-to-charge ratio (m/z). BiomiX exploits the CEU Mass Mediator database to match the m/z of the metabolomics peaks with those provided by the database, reporting the best matches for each peak. The MS1 m/z match is set by default to a 15-ppm error for positive mode, but neutral and negative modes are also available. The adducts available in the positive mode include M+H, H+2H, H+NA, H+NH4, and M+H-H2O, and in the negative mode, they include M-H, M-Cl, M+FA-H, and M-H-H2O. By default, all available MS1 databases are examined (the Human Metabolome Database [HBMD], Lipidmaps, Metlin, and Kegg), but their use is customizable. The user should carefully review both the databases and parameters according to the dataset.</p>
                <p><i>MS1 and DDA-MS/MS annotation (.mzML and .mgf files) (Untargeted HRMS):</i> Same MS1 annotation pipeline including additional steps. The fragmentation files (.mzML or .mgf) are crossed with HMDB, MassBank, and MoNA databases to match each fragmentation spectra in the samples with those available in these databases. These matches are available in the results folder. The annotation based on the fragmentation spectra MS/MS (MS2) retrieved in this step will replace those obtained based on the MS1 annotation because of its higher reliability.</p>
            </li>
            <li> 
                <p><strong>Sample type filtering:</strong> BiomiX examines the lists containing previously identified or predicted metabolites in the HBMD to filter metabolites at the same time associated with one identical peak for retaining those already identified or spotted in a type of specimen. These include plasma, urine, saliva, cerebrospinal fluid, feces, sweat, breast milk, bile and amniotic fluid samples.</p>
        </ul>
                
    <h3 id="methylomics">Methylomics</h3>

    <p>Biomix requires an expression matrix <strong>Msg</strong> where the columns <strong>s</strong> represent the samples while the rows <strong>g</strong> contain the CpG island annotation. The matrix must contain the beta-values, if not available, these can be obtained by the minfi R package. The Differential methylation analysis (DMA) is performed using the ChAMP database, providing the results of the CpG island hypermethylated and hypomethylated with a volcano plot. The threshold has been set as default to the beta value change (Δbeta) > |0.15| and p.adj < 0.05.</p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Methylomics.svg?raw=true">
    </div>

    <h3 id="interface-usage">Interface Usage</h3>
    
    <h4 id="assisted-input-modification">Assisted Input Modification</h4>
    
    <p>Click on the <strong>upload button</strong> and select the file you want to modify.</p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/INPUT_1.png?raw=true">
    </div>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/INPUT_2.png?raw=true">
    </div>

    <p>After the file selection, it is possible to decide if the selected file must be used as input for the analysis or if you desire to modify it in the BiomiX standard format.</p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/INPUT_3.png?raw=true">
    </div>

    <p>When you select <strong>"Yes"</strong> to the Pop-up <strong>"Do you want to modify the file?"</strong>, another Pop-up will appear, asking for the file separator, the header, the column containing the ID and the type of decimal separator.</p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/INPUT_4.png?raw=true">
    </div>

    <p>Then the matrix preview will appear in the "preview" tab, allowing the user to visualize if the format is the right one. If not the case it is possible to modify the table using the other tab, which contains buttons to remove specific columns and rows and eventually transpose the matrix. <strong>WARNING:</strong> The preview will include the first 10 columns and rows, so it does not represent the entire matrix.</p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/INPUT_5.png?raw=true">
    </div>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/INPUT_6.png?raw=true">
    </div>
    <h4 id="Main interface guide">Main interface guide</h4>

    <p>The main interface structure:</p>
    <li><strong>Condition.</strong> Group to analyse.</li>
    <li><strong>Control.</strong> Reference group, compared with the condition one.</li>
    <li><strong>Output.</strong> Directory where save the output results of the analysis.</li>
    <p>Omics input grid:</p>
    <li><strong>Analysis.</strong> Check the box to analyse that omic by our single omics pipelines.</li>
    <li><strong>Data type.</strong> Select the type of omics uploaded.</li>
    <li><strong>Integration.</strong> Check the box to include that omics in the MOFA integration.</li>
    <li><strong>Label.</strong> Omics name, used to name the output folder and filter the sample based on the sample names.</li>
    <li><strong>Selection.</strong> If selected filters the samples based on the sample name using a regex derived by the label name.</li>
    <li><strong>Data upload.</strong> Click on the button to upload the matrix. Here is possible to modify the matrix the the BiomiX format (.tsv). </li>
    <p>Integration grid:</p>
    <li><strong>Integration.</strong> Do you want to perform the integration with the uploaded omics data?.</li>
    <li><strong>Method.</strong> Which integration method? (Only MOFA is available for now).</li>
    <li><strong>N° Factors.</strong> How many MOFA factor do you want to calculate for the integration? (select 0 for an automatic selection).</li>
    <li><strong>Factor to explore.</strong> Which factor do you want to explore graphically? (Contributors, heatmap clustering etc...).</li>
    <li><strong>Omics overlay.</strong> Which is the minimum number of omics that a sample should have to be included in the integration?</li>
    <li><strong>Open advance option.</strong> Button to open the advance option interface</li>
    <li><strong>Open BiomiX chatbot.</strong> Button to open the BiomiX chatbot</li>
    <li><strong>Start Analysis.</strong> Button to start the analysis.</li>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/MAIN_INTERFACE_1.svg?raw=true">
    </div>
    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/MAIN_INTERFACE_2.svg?raw=true">
    </div>

    <h4 id="Guide to the advanced option interface">Guide to the advanced option interface</h4>
    <p>Advanced options interface (General section):</p>
    <li><strong>Log2FC threshold.</strong> Log2FC threshold value for significant results.</li>
    <li><strong>P.adj threshold.</strong> Adjusted p.value threshold value for significant results.</li>
    <li><strong>Gene Panel.</strong> Button to upload the gene panel for the subgrouping analysis.</li>
    <li><strong>Array type.</strong> The type of chip from which the data comes (450K/EPIC).</li>
    <li><strong>N° of genes within the panel with score > one or two.</strong> Parameters to define the positive/negative subgrouping by the gene panel. The user can modify the number of genes that must have a Z-score > one or two of the control standard deviation</li>
    <li><strong>Removal of positive control samples.</strong> If checked, the controls positive for the gene panel can be excluded from the downstream analysis.</li>
    <li><strong>N° top DE genes in heatmap</strong> Number of top genes(or metabolites) to visualize in the heatmaps.</li>
    <li><strong>Clustering distance.</strong> Clustering distance used in the heatmaps, both in subgrouping and single omics analysis.</li>
    <li><strong>Clustering methods.</strong> Clustering method used in the heatmaps, both in subgrouping and single omics analysis.</li>
    <li><strong>CPU threads.</strong> Number of CPU threads used in your PC in parallel.</li>
    <li><strong>N° MOFA input features.</strong> Number of top features used in the MOFA analysis. It is suggested to use a similar number of features in each omics</li>

    <p>Advanced options interface (Metabolomics section):</p>
    <li><strong>Metabolite annotation.</strong> Which annotation (HMDB, KEGG, compound name) are used in the uploaded annotated metabolomics matrices.</li>
    <li><strong>Ion mode.</strong> Type of ionization used in the metabolomics data acquisition.</li>
    <li><strong>M/Z Tolerance ppm.</strong> Ppm tolerance used in the MS1 annotation.</li>
    <li><strong>Adduct positive mode.</strong> Type of positive adduct generated during the metabolomics data acquisition.</li>
    <li><strong>Adduct negative mode.</strong> Type of negative adduct generated during the metabolomics data acquisition.</li>
    <li><strong>Adduct neutral mode.</strong> If checked will consider these data as neutral.</li>
    <li><strong>MS1 files upload.</strong> Button were upload the MS1 annotation (required also for the MS2) with the corresponding input number.</li>
    <li><strong>Databases MS1.</strong> Databases consulted by the MS1 annotation (By Ceu Mass Mediator).</li>
    <li><strong>mz match MS1/2.</strong> Mass charge ratio tolerance for the match between MS1 annotation and the fragmentation spectra MS2 provided within the .mzml files.</li>
    <li><strong>RT match MS1/2.</strong> Retention time tolerance for the match between MS1 annotation and the fragmentation spectra MS2 provided within the .mzml files.</li>
    <li><strong>Column.</strong> Type of column used in the liquid chromatography.</li>
    <li><strong>Databases MS2.</strong>  Databases consulted by the MS2 annotation (By Tidymass).</li>
    <li><strong>MS2 directory</strong> Directory containing the fragmentation spectra (.mzml)</li>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Advanced_1.svg?raw=true">
    </div>
    <p>Advanced options interface (Metadata section):</p>
    <li><strong>Column name</strong> Column used for the sample filtering</li>
    <li><strong>Column name</strong> Type of data in the defined column</li>
    <li><strong>Threshold/Factor selected</strong> Threshold to filter the samples (format: ">= 90" / "== 0.56") or specific condition (format: "== male" / "== treated") </li>

    
    <p>Advanced options interface (MOFA section):</p>
    <li><strong>Max iteration.</strong> Maximum number of MOFA iterations in the model training.</li>
    <li><strong>Convergence mode.</strong> Speed to train the MOFA model. The longer the more accurate.</li>
    <li><strong>Threshold contribution weight.</strong> Threshold to isolate the top MOFA factor contributors. Among the 5% top percent contributors (prefiltered by BiomiX) the user can define a contribution threshold set to 0.5 by default. </li>
    <li><strong>Type of research (Bibliography).</strong> The type of document where the contributors will be researched in Pubmed. Abstract/Title or Text word.</li>
    <li><strong>N° articles (Bibliography).</strong> Number of articles consulted by BiomiX to select article related to the significant factors. Attention, Pubmed server has a max number of request and the research of too many contributors in too many articles can drive the request interruption with the server. In this case reduce the number of articles consulted or number of contributors researched.</li>
    <li><strong>N° top contributors (Bibliography).</strong> Top contributors of each omics researched in the PubMed abstracts.</li>
    <li><strong>N° keywords extracted (Bibliography).</strong> Number of keywords extracted in each article related to significant MOFA factors. These are the keywords actively extracted selecting the most cited words of the abstract by natural language processing.</li>
    <li><strong>P.adj threshold (Pathway mining).</strong> Adjusted p.value threshold to consider significant a biological pathway.</li>
    <li><strong>Pathways shown (Pathway mining).</strong> Number of biological pathways visualized as output in the pdf reports.</li>
    <li><strong>Numerical (Clinical).</strong> Check the box to correlate the clinical data (Numeric) with the significant MOFA factor. (.tsv format and sample on the rows. Use the Assisted Input Modification if needed)</li>
    <li><strong>Binary (Clinical).</strong> Check the box to correlate the clinical data (Binary) with the significant MOFA factor. (.tsv format and sample on the rows. Use the Assisted Input Modification if needed)</li>
    <li><strong>Save advanced options (Save parameters).</strong> Click on the blue button to save the advanced parameters.</li>



    
</body>

</html>
