<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Single-omics Analysis</title>
    <style>
        body {
            font-family: 'Lato', sans-serif;
            line-height: 1.6;
            margin: 20px;
        }

        .banner {
            background-color: #3498db;
            padding: 10px;
            text-align: left;
            color: white;
        }

        .banner a {
            color: white;
            margin-right: 15px;
            text-decoration: none;
        }

        h2, h3, h4 {
            color: #3498db;
        }

        img {
            max-width: 100%;
            height: auto;
            margin: 20px 0;
        }

        .additional-features {
            margin-left: 20px;
        }

        .copy-button {
            background-color: #3498db;
            color: white;
            border: none;
            padding: 5px 10px;
            text-align: center;
            text-decoration: none;
            display: inline-block;
            font-size: 14px;
            margin-bottom: 10px;
            cursor: pointer;
            border-radius: 3px;
        }
    </style>
</head>

<body>
    
    <!-- Banner with Navigation Links -->
    <div class="banner">
       <a href="index.html">BiomiX</a>
       <a href="Installation.html">Installation</a>
       <a href="Tutorial.html">Tutorial</a> 
       <a href="Single-omics_analysis.html">Single-omics analysis</a>
       <a href="Integration_analysis.html">Integration</a>
       <a href="Biological_interpretation.html">Biological Interpretation</a>
       <a href="about_us.html">About us</a>
    </div>

    <h3 id="General-guidelines">General guidelines</h3>
    <p>BiomiX was developed to compare two groups using single and multiple omics. To work properly, the input file must follow a specific format and criteria.</p>
    <p><strong>Metadata: </strong></p>
    <p>BiomiX requires uploading a metadata file in the BiomiX launcher. The first column must contain the sample ID, while the others contain other sample variables.</p>
    <li><strong>"ID" column (mandatory):</strong> This is the first column and contains the sample ID.</li>
    <li><strong>"CONDITION" column (mandatory):</strong> This column must contain the sample condition. These conditions will be available for selection in the main interface.</li>
    <li><strong>"SAMPLE_TYPE" column (optional):</strong> This column is designed for specific conditions. Suppose the input matrix contains samples from different tissues or cells (convoluted matrix), and this source is also defined in the sample ID (e.g., ID = 25658_BLymphocyte, 25658_GlialCells). In that case, this column allows analysis to be performed only on data from the selected tissues or cells.</li>
    <li><strong>"GENDER" column (optional): </strong> This column can be used to adjust the LIMMA or DESeq2 model for transcriptomics data.</li>
    <li><strong>"AGE" column (optional):</strong> This column can also be used to adjust the LIMMA or DESeq2 model for transcriptomics data.</li>
 
<p>Any other metadata column can be used to filter the data via the Metadata section in the advanced options or for the PCA and UMAP preview-QC visualization.</p>
    
    <h3 id="Matrix"><p><strong>Matrix: </strong></p></h3>

    <p>BiomiX requires an expression matrix where the columns represent the samples and the rows contain the variables.</p>
    <p><strong>Omics differences: </strong></p>
      
    <li><strong>In transcriptomics,</strong> the variables (genes) should be named using GENE SYMBOL or ENSEMBL ID.</li>
    <li><strong>In metabolomics,</strong> variables (metabolites) should be named using KEGG, HMDB, or official compound names. If the peaks are not annotated, the format should be "peak_[number]," especially for MS1 and MS2 annotation.</li>
    <li><strong>In methylomics, </strong> variables must use the EPIC or 450K array code. </li>
    <li><strong>In Undefined omics, </strong> any variable name is accepted.</li>

    
    <h3 id="transcriptomics">Transcriptomics</h3>

    <h4 id="Single-Omics-Analysis">Single Omics Analysis</h4>
    <p>The transcriptomics single omics analysis identifies differentially expressed genes (DEGs) along with their statistics (Log2FC and adjusted p-values), and provides visualizations such as volcano plots and heatmaps. If <strong>SEX</strong> and <strong>GENDER</strong> columns are included in the metadata, they will be used to adjust the DESeq2 or Limma model. The default thresholds for DEGs are set to Log2FC > 0.5 and adjusted p-value < 0.05.</p>

    <p>Biological process enrichment is explored using the R version of EnrichR, with options to export data for external tools such as <strong>GSEA</strong> <a href="https://www.gsea-msigdb.org/gsea/downloads.jsp"> Link </a> and <strong>EnrichR</strong> <a href="https://maayanlab.cloud/Enrichr/"> Link </a>. Within the Preview-QC visualization, users can choose to transform the data: if transformed, DGE is performed with Limma; if not, and the expression matrix contains raw counts, DESeq2 is used instead. The transformed data are then used as input for MOFA integration. To optimize analysis speed, only the most variable genes are selected (5000 by default, customizable in the MOFA section of the advanced options).</p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Transcriptomics.svg?raw=true">
    </div>

    <div class="additional-features">
        <h4>Additional Features</h4>

        <ul>
            <li>
                <strong>Gene Panel subgrouping:</strong> It is possible to provide a panel of genes within <strong>GENE_PANEL</strong> file, in the <strong>General section of the advanced options</strong> The genes must be in Gene symbol format and tab-delimited (by enter button).
                <pre><code>GENES_FOR_SUBPOPULATION
SIGLEC1
IFIT3
IFI6
LY6E
MX1
USP18
OAS3
IFI44L
</code></pre>
               These genes will generate a heatmap displaying the expression standard deviation of each specific gene across samples. This enables classification of condition/disease samples into positive and negative groups, while BiomiX performs DGE analysis to compare the chosen controls with these two subgroups
            </li>
            <li>
            <strong>Clinical or Biological Marker Validation:</strong> Adding a column named <strong>MARKER</strong> in the metadata is possible to classify samples based on clinical or biological marker measurements. This marker classification can then be compared to gene panel subgrouping to observe similarities or differences.
            </li>
        </ul>

        <div align="center">
            <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Heatmap_BLymphocytes_CTRL_vs_SLE2.svg?raw=true">
        </div>

        <div align="center">
            <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/enrichR.svg?raw=true">
        </div>

        <div align="center">
            <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/GSEA2.svg?raw=true">
        </div>
    </div>

    <h3 id="metabolomics">Metabolomics</h3>

    <p>The matrix's peak or metabolite signals are first visualized by preview-QC to decide if and how to transform the data. The transformed signals are then compared among the chosen groups by calculating Log2FC as the log2 ratio of their median peak signals. P-values are calculated using the non-parametric Mann–Whitney test and corrected using the FDR method. The Volcano plot and heatmap display significant metabolites or peaks that are increased or decreased. <strong>Warning:</strong> Please ensure the data matrix is curated to remove variables without biological relevance (e.g., contaminants, poorly reproducible signals) and that instrumental normalization is performed before uploading to BiomiX. All transformed peak or metabolite signals are included for MOFA integration.</p>

    <p>To reveal changes in biological pathways, BiomiX uses methPath v1.0.5 from tidymass v1.0.8 to detect enrichment in metabolic processes among significantly changed metabolites. For deeper analysis, it also prepares ready-to-copy files for input into MetaboAnalyst, one of the most advanced online tools for metabolomics analysis, supporting enrichment and metabolite set enrichment analysis. If transcriptomic results are available in BiomiX, the platform will automatically integrate metabolomics and transcriptomics data to create files ready for MetaboAnalyst's joint pathway and network analysis.</p>

    <p>If you have only raw .mzML files, we strongly recommend using the R pipeline by Á. Fernández-Ochoa (https://doi.org/10.3390/metabo10010028) or MetaboAnalyst to generate the peak signals matrix.</p>


    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Metabolomics.svg?raw=true">
    </div>

    <div class="additional-features">
        <h4>Additional Features</h4>

        <p>BiomiX is flexible, supporting the analysis of both targeted (annotated) and untargeted metabolomics data, with the option to analyze untargeted data with or without annotation. The untargeted annotation is based on MS1 annotation (mass/charge ratio “m/z” and retention time values) or MS2 data (MS1 annotation plus raw MS2 fragmentation files in mzML or .mgf format).</p>

        <ul>
            <li>
                <strong>Peaks automatic annotation:</strong>
                <p><i><strong>Targeted metabolomics:</strong></i> This option supports metabolomics data obtained from any analytical platform such as LC-MS, GC-MS, CE-MS or NMR, provided the metabolites' biological identities are available. Despite the annotation here is not performed the choice of the proper metabolite identifier (HMDB or KEGG) is required for biological pathway analysis. Also the Iupac compound name option is available but will not perform any biological pathway analysis.</p>
                <p><i><strong>MS1 annotation (Untargeted HRMS: High-resolution mass spectrometry):</strong></i> Users can upload MS1 files, containing mass-to-charge ratio (m/z). BiomiX exploits the CEU Mass Mediator database to match the m/z of the metabolomics peaks with those provided by the database, reporting the best matches for each peak. The MS1 m/z match is set by default to a 15-ppm error for positive mode, but neutral and negative modes are also available. The adducts available in the positive mode include M+H, H+2H, H+NA, H+NH4, and M+H-H2O, and in the negative mode, they include M-H, M-Cl, M+FA-H, and M-H-H2O. By default, all available MS1 databases are examined (the Human Metabolome Database [HBMD], Lipidmaps, Metlin, and Kegg), but their use is customizable. The user should carefully review both the databases and parameters according to the dataset.</p>
                <p><i><strong>MS1 and DDA-MS/MS annotation (.mzML and .mgf files) (Untargeted HRMS):</strong></i> Same MS1 annotation pipeline including additional steps. The fragmentation files (.mzML or .mgf) are crossed with HMDB, MassBank, and MoNA databases to match each fragmentation spectra in the samples with those available in these databases. These matches are available in the results folder. The annotation based on the fragmentation spectra MS/MS (MS2) retrieved in this step will replace those obtained based on the MS1 annotation because of its higher reliability.</p>
            </li>
            <li> 
                <p><strong>Sample type filtering:</strong> BiomiX examines the lists containing previously identified or predicted metabolites in the HBMD to filter metabolites at the same time associated with one identical peak for retaining those already identified or spotted in a type of specimen. These include plasma, urine, saliva, cerebrospinal fluid, feces, sweat, breast milk, bile and amniotic fluid samples.</p>
        </ul>
                
    <h3 id="methylomics">Methylomics</h3>

    <p>BiomiX performs a Differential methylation analysis using the ChAMP packae, providing the CpG island Δbeta value, the p-adjusted corrected by FDR and a summarizing volcano plot and heatmap. The threshold has been set as the default to the beta value change (Δbeta) > |0.15| and p.adj corrected by the FDR method < 0.05. </p>

    <div align="center">
        <img src="https://github.com/IxI-97/IxI-97.github.io/blob/main/Methylomics.svg?raw=true">
    </div>

    
</body>

</html>
